src/openms/include/OpenMS/FORMAT/FileTypes.h

```diff
@@ -95,11 +95,28 @@ namespace OpenMS
 
     enum class FileProperties
     {
-      READABLE,
-      WRITEABLE,
-      PROVIDES_MS1,
-      PROVIDES_MS2,
-      SIZE_OF_FILEPROPERTIES
+      READABLE,                         // SOMETHING in OpenMS can read this (it doesn't have to be in FileHandler though)
+      WRITEABLE,                        // SOMETHING in OpenMS can write this (it doesn't have to be in FileHandler though) 
+      PROVIDES_MS1,                     // File CAN contain spectra with no parent
+      PROVIDES_MS2,                     // File CAN contain spectra that have a parent
+      PROVIDES_LOADSPECTRUM,            // All of the PROVIDES_x properties correspond to which FileHandlers are implemented for a file type.
```

Comment:
What are these methods?
Do you really want to model what functions the filetypehandler provides? Do you need this for some method dispatch?
This look like bad design at first glance (at least got some alarm clocks ringing ;) )

Reply:
I added these as part of taking inventory of what load and store operations we have implemented for each file type. Whether there is actually a handler is already checked internally. If you think this is unnecessary I can take them back out, but then the READABLE or WRITEABLE properties don't give any info as to how the files are readable or writeable.

Reply:
are these used somewhere?

Reply:
Not yet. I've been trying to keep this as compact as possible, as it's already a lot of code. I'm more than happy to add these in a separate PR in the future bu @cbielow's suggestion above was to complete the properties.

Reply:
Yeah, I think this needs to go. Inventories are nearly always a sign of non-local code and bad design, which we should prevent here because we need to keep this part extensible.

READABLE or WRITEABLE should just tell you if a file is readable/writable or both. Can you point me where you need more information for proper reading / writing?

Reply:
Okay. I can get rid of them. The one the comes to mind is CACHEDMZML, where we have loading and storing for it, but they aren't yet in the FileHandler class, so it's unclear without digging through the CachedMZML classes how they get handled.

Reply:
@cbielow can you clarify if you meant this with completion?
Why not have e.g, PROVIDES_QC and WRITABLE I think this makes it clear that it is a file format that we can write to and that it provides QC ?

Reply:
> Okay. I can get rid of them. The one the comes to mind is CACHEDMZML, where we have loading and storing for it, but they aren't yet in the FileHandler class, so it's unclear without digging through the CachedMZML classes how they get handled.

Not sure if I understand this part. Do you want to use it through the same FileHandler interface? Then it probably needs to be added there.

Reply:
> Yeah, I think this needs to go. Inventories are nearly always a sign of non-local code and bad design, which we should prevent here because we need to keep this part extensible.
> 
> READABLE or WRITEABLE should just tell you if a file is readable/writable or both. Can you point me where you need more information for proper reading / writing?
Oh, I like this a lot better anyway. I'm just going to roll with that.